File: Extra Credit/Third Lightbulb/Fibonacci.java
/**
 * Christopher Oh
 * SJSU CS 146
 * Bonus Questions: "Third lightbulb's the charm"
 */

public class Fibonacci {
    /**
     * Computes the nth number in the Fibonacci Sequence
     * Note: Counting starts at index 0
     * @param n
     * @return nth number of Fibonacci Sequence
     */
    public static int nthFibonacci (int n) {
        if (n < 0) {
            System.out.println("Error, n must be a non-negative integer");
            return -1;
        }

        int a = 0;
        int b = 1;

        for (int i = 1; i <= n; i++) {
            int temp = a;
            a = b;
            b = temp + b;
        }
        return a;
    }
    
    public static void main(String[] args) {
        System.out.println(nthFibonacci(17));
    }
}

File: Extra Credit/Third Lightbulb/Fibonacci.py
# Christopher Oh
# SJSU CS 146
# Bonus Questions: "Third lightbulb's the charm"
# Fibonacci Sequence

def nth_Fibonacci (n):
    if n < 0:
        print("Error: n must be a non-negative integer")
        return -1

    a = 0
    b = 1

    for i in range(1, n + 1):
        temp = a
        a = b
        b = temp + b

    return a


def main():
    print(nth_Fibonacci(17))


if __name__ == '__main__':
    main()
